lib/src/pub.dart

```diff
@@ -87,6 +108,41 @@ class Pubspec {
       this._contents,
       this._yamlMap);
 
+  static Future<Pubspec> init() {
+    var completer = new Completer();
+    var packageRoot = p.current;
+    var pubspecPath = p.join(packageRoot, _PUBSPEC);
+    var exp = new RegExp(r"(-|\.).*$");
+    var name = p.basename(packageRoot).replaceAll(exp, "");
+    var contents = "name: $name";
+    var yaml = loadYamlNode(contents, sourceUrl: pubspecPath);
+    var _pubspec = new Pubspec(pubspecPath, contents, yaml);
+
+    _pubspec.version = new Version.parse('0.1.0');
+
+    runZoned(() {
+      checkHasGit()
```

Comment:
Zones should not be necessary, you can just do something like:

```dart
return checkHasGit().then(...).then(...).catchError((error) {
  _pubspec.author = '';
  return _pubspec;
});
```

That is to say you can just return the a new Future value from the callbacks passed to `then` or `catchError` instead of using Zones or Completers.